### PR TITLE
TIMOB-16080: Added New sound property for views

### PR DESCRIPTION
**Pull request by antw12**
**JIRA:** 
https://jira.appcelerator.org/browse/TIMOB-16080

**Description:**
Added a new property PROPERTY_TOUCH_SOUND_ENABLED allowing a user to set sound effects for views on or off.

Default for property is true

**Test Case**
```
var win = Ti.UI.createWindow({
    backgroundColor: 'black',
});

var view = Ti.UI.createView({
    soundEffectsEnabled : false,
    backgroundColor : "Black"
});
win.add(view);
win.open();
```

### CODE REVIEW

File: android/titanium/src/java/org/appcelerator/titanium/TiC.java [hunk 1]
```diff
@@ -2746,6 +2746,11 @@
 	/**
 	 * @module.api
 	 */
+	public static final String PROPERTY_TOUCH_SOUND_ENABLED = "soundEnabled";
```
Comment (hansemannn) [inline android/titanium/src/java/org/appcelerator/titanium/TiC.java]: We should name it the same (`soundEffectsEnabled`), which would be mapped to `PROPERTY_SOUND_EFFECTS_ENABLED`.

File: android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java [hunk 2]
```diff
@@ -1726,8 +1726,14 @@ protected void registerForTouch(final View touchable)
 				touchable.setEnabled(false);
 			}
 		}
+		//Checking and setting touch sound for view
+		if (proxy.hasProperty(TiC.PROPERTY_TOUCH_SOUND_ENABLED)) {
+		 boolean soundEnabled = TiConvert.toBoolean(proxy.getProperty(TiC.PROPERTY_TOUCH_SOUND_ENABLED), true);
```
Comment (hansemannn) [inline android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java]: Indentation incorrect.

File: android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java [hunk 3]
```diff
@@ -1726,8 +1726,14 @@ protected void registerForTouch(final View touchable)
...
+			if (!soundEnabled) {
```
Comment (hansemannn) [inline android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java]: If this is `false` first, setting it to `true` will skip every time. Rather call the setter directly.

**Review by hansemannn — CHANGES_REQUESTED** [review]

File: android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java (hunk 3 above)
Comment (antw12) [inline android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java]: It is set to true as default as that is the native behavior, I tested it and it works so i am not sure, if I am still wrong please let me know

File: android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java [hunk 4]
```diff
@@ -1719,15 +1719,20 @@ protected void registerForTouch(final View touchable)
...
 		}
-
 		if (proxy.hasProperty(TiC.PROPERTY_TOUCH_ENABLED)) {
 			boolean enabled = TiConvert.toBoolean(proxy.getProperty(TiC.PROPERTY_TOUCH_ENABLED), true);
 			if (!enabled) {
 				touchable.setEnabled(false);
 			}
 		}
+		//Checking and setting touch sound for view
+		if (proxy.hasProperty(TiC.PROPERTY_SOUND_EFFECTS_ENABLED)) {
```
Comment (garymathews) [inline android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java]:
```Java
if (proxy.hasProperty(TiC.PROPERTY_SOUND_EFFECTS_ENABLED)) {
	boolean soundEnabled = TiConvert.toBoolean(proxy.getProperty(TiC.PROPERTY_SOUND_EFFECTS_ENABLED), true);
	touchable.setSoundEffectsEnabled(soundEnabled);
}
```

**Review by garymathews — CHANGES_REQUESTED** [review]

File: android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java [hunk 5]
```diff
@@ -116,6 +116,7 @@ public TiViewProxy()
 		pendingAnimationLock = new Object();
 
 		defaultValues.put(TiC.PROPERTY_TOUCH_ENABLED, true);
+		defaultValues.put(TiC.PROPERTY_SOUND_EFFECTS_ENABLED, true);
```
Comment (garymathews) [inline android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java]: This is necessarily needed since it's already enabled by default

**Comment by antw12** [comment]
for the PROPERTY_TOUCH_ENABLED should i change it to match, as it is set up the same with default as true?

**Comment by garymathews** [comment]
@antw12 Yes, as that will not allow you to set `touchEnabled` back to `true`. Nice catch 👍

**Comment by garymathews** [comment]
CR: **PASS** (note: try not to change the formatting of code you are not changing)
FT: **PASS**

**Comment by ssjsamir** [comment]
FR Passed, was able to set `soundEffectsEnabled` to `true` or `false` and can confirm if it was set to to `true` sound effects can be heard and if set to `false` sound effects can not be heard. 

Steps I took to test:
+ Copy the code provided in the description
+ Ran the program
+ Was not able to hear the sound effects 
+ Edited the code and set `soundEffectsEnabled` to `true` 
+ Was able to hear the sound effects 

Environment

Appcelerator Command-Line Interface, version 6.1.0
Google Pixel XL (Android 7.1.1)
Operating System Name: Mac OS X El Capitan
Operating System Version: 10.11.6
Node.js Version: 4.6.0
npm: 4.2.8
Xcode: 8.2
Appcelerator Studio: 4.8.1.201612050850

**Pull request merged**